mayastor/src/grpc/mayastor_grpc.rs

```diff
@@ -254,23 +305,38 @@ impl mayastor_server::Mayastor for MayastorSvc {
             async move {
                 let args = request.into_inner();
                 info!("{:?}", args);
-                let rx = rpc_submit::<_, _, LvsError>(async move {
-                    if let Some(pool) = Lvs::lookup(&args.name) {
-                        // Remove pool from current config and export to file.
-                        // Do this BEFORE we actually destroy the pool.
-                        let mut config = PoolConfig::capture();
-                        config.delete(&args.name);
-                        config.export().await;
-
-                        pool.destroy().await?;
+                let resp = match PoolBackend::try_from(args.pooltype)? {
```

Comment:
I am in a doubt whether we should use pool type when removing the pool, as we can't have 2 pools of different types with the same name. IMHO, we should use only pool name as input and perform pool type traversal inside gRPC call, hiding it from user. That will bring us benefits when calling our methods via REST API, as user normally uses pool name as its ID when calling 'DEL /pools/:id', which then gets simply passed to our gRPC without performing one extra call for resolving pool type.

Reply:
Makes sense. The workflow would be something like 
```
if lvs_lookup(name){
  lvs_del_pool(name)
  return
}

if lvm_enabled && lvm_lookup(name){
  lvm_del_pool(name)
  return
}
```
?